dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/common/DefaultIdentifiableObjectManager.java

```diff
@@ -1085,6 +1085,14 @@ public void evict( Object object )
             .build();
     }
 
+    @Override
+    @Transactional( readOnly = true )
+    public List<String> getUidsCreatedBefore( Class<? extends IdentifiableObject> klass, Date date )
+    {
+        IdentifiableObjectStore<IdentifiableObject> store = getIdentifiableObjectStore( klass );
+        return store.getUidsCreatedBefore( date );
```

Comment:
Unfortunately I think we need a null check for the store variable here, as there might not be a store defined (this is not a great solution overall):

```
        if ( store == null )
        {
            return new ArrayList<>();
        }
```